java/rocksjni/transaction.cc

```diff
@@ -631,27 +614,23 @@ void txn_write_kv_parts_helper(JNIEnv* env,
                                const jint& jvalue_parts_len) {
   assert(jkey_parts_len == jvalue_parts_len);
 
-  rocksdb::Slice key_parts[jkey_parts_len];
-  rocksdb::Slice value_parts[jvalue_parts_len];
-  std::tuple<jbyteArray, jbyte*, jobject> jkey_parts_to_free[jkey_parts_len];
-  std::tuple<jbyteArray, jbyte*, jobject>
-      jvalue_parts_to_free[jvalue_parts_len];
+  auto key_parts = std::vector<rocksdb::Slice>();
+  auto value_parts = std::vector<rocksdb::Slice>();
+  auto jparts_to_free = std::vector<std::tuple<jbyteArray, jbyte*, jobject>>();
 
   // convert java key_parts/value_parts byte[][] to Slice(s)
   for (jsize i = 0; i < jkey_parts_len; ++i) {
     const jobject jobj_key_part = env->GetObjectArrayElement(jkey_parts, i);
     if (env->ExceptionCheck()) {
       // exception thrown: ArrayIndexOutOfBoundsException
-      free_key_value_parts(env, jkey_parts_len, jkey_parts_to_free,
-                           jvalue_parts_to_free);
+      free_key_parts(env, jparts_to_free);
```

Comment:
Can you rename `free_key_parts` to just `free_parts` or similar, as it now deals with both keys and values.

Reply:
Also where is that function? I don't see it in your PR

Reply:
I kept original function free_key_parts. It was there two times so I used first and deleted second. I can rename it. No problem.

Reply:
Ah okay, thanks yes, please rename